library/src/schemas/object/object.ts

```diff
@@ -112,10 +113,8 @@ export function object<TObjectShape extends ObjectShape>(
       for (const [key, schema] of Object.entries(object)) {
         try {
           const value = (input as Record<string, unknown>)[key];
-          output[key] = schema.parse(value, {
-            ...info,
-            path: getCurrentPath(info, { schema: 'object', input, key, value }),
-          });
+          output[key] = schema.parse(value, getParseInfoWithPath(info,
+            getCurrentPath(info, { schema: 'object', input, key, value })))
```

Comment:
Note that this form is kind of weird now, I would refactor  this into something like the following:
```typescript
item.parse(value, getParseInfoWithAppendedPath(info, {
  schema: 'object',
  input,
  key,
  value,
}))
```
ideas welcome